internal/stream/stream.go

```diff
@@ -184,9 +187,38 @@ func (s *Stream) StartReader(reader Reader) {
 
 	sr.start()
 
-	for _, sm := range s.streamMedias {
+	for m, sm := range s.streamMedias {
 		for _, sf := range sm.formats {
 			sf.startReader(sr)
+			if m.Type == "video" {
+				cb := sf.runningReaders[sr]
+				if cb == nil {
+					continue
+				}
+
+				go func() {
```

Comment:
Question: For this goroutine, if there are lots of CachedUnits, do we need to worry about context cancellation? Like if the stream is disconnected while we're asleep, can we keep calling the callback afterwards?

Reply:
yeah it would be good to check the context if there's one around:

```go
select {
  case <-something.Ctx.Done():
     return
  default:
     // the original loop body
}
```

Reply:
I am now using `sr.push(func() error {` and the error seems to be handled there!